src/Dobble/DeckGenerator.php

```diff
@@ -74,4 +84,171 @@ public function generate()
+    }
+
+    /**
+     * @param $modulo Integer modulo
+     * @param $matrix Matrix of unique values for given integer modulo
+     * @param $infinity List of points at infinity for given integer modulo
+     * @param $a Draw a line for f(x) = ax + b
+     * @param $b Draw a line for f(x) = ax + b
+     * @param $f For $f = 5, draw a line for f(5). If given, $a and $b will be ignored
+     * @return array All points through which pass the line
+     */
+    private function fetchLine($modulo, $matrix, $infinity, $a, $b, $f = null)
+    {
+        $coords = []; // Coordinates of all points
+        $points = [];
+
+        for($i = 0; $i < $modulo; ++$i) {
+            if ($f === null) {
+                // f(x) = ax + b
+                // We apply the modulo operation to results because
+                // There is not integers but modulo integers
+                $x = $i % $modulo;
+                $y = ($a * $i + $b) % $modulo;
+            } else {
+                $x = $f;
+                $y = $i % $modulo;
+            }
+
+            // Add the calculated coordinates to the list
+            array_push($coords, [$x, $y]);
+        }
+
+        // Fetching the value for all points
+        foreach($coords as $coord) {
+            $x = $coord[0];
+            $y = $coord[1];
+            array_push($points, $matrix[$y][$x]);
+        }
+
+        // Add the last point at infinity
+        if ($f === null) {
+            array_push($points, $infinity[$a + 1]);
+        } else {
+            array_push($points, $infinity[0]);
+        }
+
+        return $points;
+    }
+
+    /**
+     * @param $modulo Integer modulo
+     * @param $matrix
+     * @param $infinity
+     * @return array All lines of the projective plan
+     */
+    private function fetchAllLines($modulo, $matrix, $infinity)
+    {
+        $lines = [[]];
+
+        // The first line is the line which pass to all points at infinity
+        foreach($infinity as $point) {
+            array_push($lines[0], $point);
+        }
+
+        // Fetch all verticals lines
+        for($i = 0; $i < $modulo; ++$i) {
+            $line = $this->fetchLine($modulo, $matrix, $infinity, 0, 0, $i);
+            array_push($lines, $line);
+        }
+
+        // Finally, fetch all lines for f(x) = ax + b
+        for($a = 0; $a < $modulo; ++$a) {
+            for($b = 0; $b < $modulo; ++$b) {
+                $line = $this->fetchLine($modulo, $matrix, $infinity, $a, $b);
+                array_push($lines, $line);
+            }
+        }
+
+        return $lines;
+    }
+
+    /**
+     * @description Apply a quadric equation to resolve the formula : modulo = x*x + x + 1
+     * @param $nbSymbols Amount of symbols in all projective plan
+     * @return int Integer modulo
+     */
+    private function calculateModulo($nbSymbols)
+    {
+        $delta = -3 + 4 * $nbSymbols;
+
+        if ($delta < 0) {
+            return false;
+        } elseif ($delta == 0) {
```

Comment:
Not required. If delta is null, then the two roots would be equal, and then the following of the method would be enough.